Index: app/src/main/java/com/example/shopeaze/SecondFragment.java
===================================================================
diff --git a/app/src/main/java/com/example/shopeaze/SecondFragment.java b/app/src/main/java/com/example/shopeaze/SecondFragment.java
deleted file mode 100644
--- a/app/src/main/java/com/example/shopeaze/SecondFragment.java	(revision 4f3b9dcc605b445b297203dda541b5e2d925738d)
+++ /dev/null	(revision 4f3b9dcc605b445b297203dda541b5e2d925738d)
@@ -1,47 +0,0 @@
-package com.example.shopeaze;
-
-import android.os.Bundle;
-import android.view.LayoutInflater;
-import android.view.View;
-import android.view.ViewGroup;
-
-import androidx.annotation.NonNull;
-import androidx.fragment.app.Fragment;
-import androidx.navigation.fragment.NavHostFragment;
-
-import com.example.shopeaze.databinding.FragmentSecondBinding;
-
-public class SecondFragment extends Fragment {
-
-    private FragmentSecondBinding binding;
-
-    @Override
-    public View onCreateView(
-            LayoutInflater inflater, ViewGroup container,
-            Bundle savedInstanceState
-    ) {
-
-        binding = FragmentSecondBinding.inflate(inflater, container, false);
-        return binding.getRoot();
-
-    }
-
-    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
-        super.onViewCreated(view, savedInstanceState);
-
-        binding.buttonSecond.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                NavHostFragment.findNavController(SecondFragment.this)
-                        .navigate(R.id.action_SecondFragment_to_FirstFragment);
-            }
-        });
-    }
-
-    @Override
-    public void onDestroyView() {
-        super.onDestroyView();
-        binding = null;
-    }
-
-}
\ No newline at end of file
Index: app/src/main/res/layout/fragment_second.xml
===================================================================
diff --git a/app/src/main/res/layout/fragment_second.xml b/app/src/main/res/layout/fragment_second.xml
deleted file mode 100644
--- a/app/src/main/res/layout/fragment_second.xml	(revision 4f3b9dcc605b445b297203dda541b5e2d925738d)
+++ /dev/null	(revision 4f3b9dcc605b445b297203dda541b5e2d925738d)
@@ -1,35 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<androidx.core.widget.NestedScrollView xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    tools:context=".SecondFragment">
-
-    <androidx.constraintlayout.widget.ConstraintLayout
-        android:layout_width="match_parent"
-        android:layout_height="match_parent"
-        android:padding="16dp">
-
-        <Button
-            android:id="@+id/button_second"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
-            android:text="@string/previous"
-            app:layout_constraintBottom_toTopOf="@id/textview_second"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toTopOf="parent" />
-
-        <TextView
-            android:id="@+id/textview_second"
-            android:layout_width="wrap_content"
-            android:layout_height="wrap_content"
-            android:layout_marginTop="16dp"
-            android:text="@string/lorem_ipsum"
-            app:layout_constraintBottom_toBottomOf="parent"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toBottomOf="@id/button_second" />
-    </androidx.constraintlayout.widget.ConstraintLayout>
-</androidx.core.widget.NestedScrollView>
\ No newline at end of file
Index: app/src/main/res/layout/activity_login.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/res/layout/activity_login.xml b/app/src/main/res/layout/activity_login.xml
--- a/app/src/main/res/layout/activity_login.xml	(revision 4f3b9dcc605b445b297203dda541b5e2d925738d)
+++ b/app/src/main/res/layout/activity_login.xml	(date 1690497893546)
@@ -29,8 +29,8 @@
         </com.google.android.material.textfield.TextInputEditText>
     </com.google.android.material.textfield.TextInputLayout>
 
-
-
+<!--password:
+-->
     <com.google.android.material.textfield.TextInputLayout
         android:layout_width="match_parent"
         android:layout_height="wrap_content">
Index: app/src/main/res/values/strings.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/res/values/strings.xml b/app/src/main/res/values/strings.xml
--- a/app/src/main/res/values/strings.xml	(revision 4f3b9dcc605b445b297203dda541b5e2d925738d)
+++ b/app/src/main/res/values/strings.xml	(date 1690673540847)
@@ -22,4 +22,5 @@
     <string name="don_t_have_an_account_sign_up">Don\'t have an account? Sign Up!</string>
     <string name="click_here_to_create_an_account">Click here to create an account</string>
     <string name="create_an_account">Create An Account</string>
+    <string name="log_out">Log Out</string>
 </resources>
\ No newline at end of file
Index: app/src/main/res/navigation/nav_graph.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/res/navigation/nav_graph.xml b/app/src/main/res/navigation/nav_graph.xml
--- a/app/src/main/res/navigation/nav_graph.xml	(revision 4f3b9dcc605b445b297203dda541b5e2d925738d)
+++ b/app/src/main/res/navigation/nav_graph.xml	(date 1690680234947)
@@ -2,32 +2,32 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:id="@+id/nav_graph"
-    app:startDestination="@id/FirstFragment">
+    app:startDestination="@id/WelcomeScreen">
 
     <fragment
-        android:id="@+id/FirstFragment"
+        android:id="@+id/WelcomeScreen"
         android:name="com.example.shopeaze.WelcomeScreen"
         android:label="@string/first_fragment_label"
-        tools:layout="@layout/fragment_welcome">
+        tools:layout="@layout/activity_main">
 
         <action
             android:id="@+id/action_FirstFragment_to_ShopperFragment"
-            app:destination="@id/ShopperFragment" />
+            app:destination="@id/SecondFragment" />
 
-        <action
+        <!-- <action
             android:id="@+id/action_FirstFragment_to_OwnerFragment"
-            app:destination="@id/OwnerFragment" />
+            app:destination="@id/OwnerFragment" /> -->
     </fragment>
 
     <fragment
         android:id="@+id/SecondFragment"
-        android:name="com.example.shopeaze.SecondFragment"
+        android:name="com.example.shopeaze.Login"
         android:label="@string/second_fragment_label"
-        tools:layout="@layout/fragment_second">
+        tools:layout="@layout/activity_login">
 
         <action
             android:id="@+id/action_SecondFragment_to_FirstFragment"
-            app:destination="@id/FirstFragment" />
+            app:destination="@id/WelcomeScreen" />
     </fragment>
 
     <!-- Add the new fragments for Shopper and Owner screens -->
@@ -45,4 +45,4 @@
         android:label="@string/owner_fragment_label"
         tools:layout="@layout/fragment_owner" />
     -->
-</navigation>
+</navigation>
\ No newline at end of file
Index: app/src/main/java/com/example/shopeaze/WelcomeScreen.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/com/example/shopeaze/WelcomeScreen.java b/app/src/main/java/com/example/shopeaze/WelcomeScreen.java
--- a/app/src/main/java/com/example/shopeaze/WelcomeScreen.java	(revision 4f3b9dcc605b445b297203dda541b5e2d925738d)
+++ b/app/src/main/java/com/example/shopeaze/WelcomeScreen.java	(date 1690679526809)
@@ -43,8 +43,8 @@
         ownerButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                NavHostFragment.findNavController(WelcomeScreen.this)
-                        .navigate(R.id.action_FirstFragment_to_OwnerFragment);
+                //    NavHostFragment.findNavController(WelcomeScreen.this)
+                //            .navigate(R.id.action_FirstFragment_to_OwnerFragment);
             }
         });
     }
Index: app/src/main/java/com/example/shopeaze/SignUp.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/com/example/shopeaze/SignUp.java b/app/src/main/java/com/example/shopeaze/SignUp.java
--- a/app/src/main/java/com/example/shopeaze/SignUp.java	(revision 4f3b9dcc605b445b297203dda541b5e2d925738d)
+++ b/app/src/main/java/com/example/shopeaze/SignUp.java	(date 1690677620734)
@@ -1,90 +1,113 @@
 package com.example.shopeaze;
 
 import androidx.annotation.NonNull;
-import androidx.appcompat.app.AppCompatActivity;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
 
 import android.content.Intent;
 import android.os.Bundle;
 import android.text.TextUtils;
-import android.util.Log;
+import android.view.LayoutInflater;
 import android.view.View;
+import android.view.ViewGroup;
 import android.widget.Button;
 import android.widget.ProgressBar;
 import android.widget.TextView;
 import android.widget.Toast;
-import com.google.firebase.auth.FirebaseAuth;
-import com.google.firebase.auth.AuthResult;
+
 import com.google.android.gms.tasks.OnCompleteListener;
 import com.google.android.gms.tasks.Task;
-
 import com.google.android.material.textfield.TextInputEditText;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
 import com.google.firebase.auth.FirebaseUser;
 
-public class SignUp extends AppCompatActivity {
+public class SignUp extends Fragment {
+
     TextInputEditText editTextEmail, editTextPassword;
     Button buttonSignUp;
     FirebaseAuth mAuth;
     ProgressBar progressBar;
     TextView textView;
+
     @Override
     public void onStart() {
         super.onStart();
-        // Check if user is alr signed in (non-null)
+        mAuth = FirebaseAuth.getInstance();
+        // Check if user is already signed in (non-null)
         FirebaseUser currentUser = mAuth.getCurrentUser();
         if(currentUser != null){
-            Intent intent = new Intent(getApplicationContext(), MainActivity.class);
+            Intent intent = new Intent(getActivity(), MainActivity.class);
             startActivity(intent);
-            finish();
+            getActivity().finish();
         }
     }
+
+    @Nullable
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_sign_up);
-        mAuth = FirebaseAuth.getInstance();
-        editTextEmail = findViewById(R.id.email);
-        editTextPassword = findViewById(R.id.password);
-        buttonSignUp = findViewById(R.id.btn_signup);
-        progressBar = findViewById(R.id.progressBar);
-        textView = findViewById(R.id.loginNow);
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.activity_sign_up, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        editTextEmail = view.findViewById(R.id.email);
+        editTextPassword = view.findViewById(R.id.password);
+        buttonSignUp = view.findViewById(R.id.btn_signup);
+        progressBar = view.findViewById(R.id.progressBar);
+        textView = view.findViewById(R.id.loginNow);
+
         textView.setOnClickListener(new View.OnClickListener(){
             @Override
-            public void onClick(View view){
-                Intent intent = new Intent(getApplicationContext(), Login.class);
+            public void onClick(View v){
+                Intent intent = new Intent(getActivity(), Login.class);
                 startActivity(intent);
-                finish();
+                getActivity().finish();
             }
         });
 
         buttonSignUp.setOnClickListener(new View.OnClickListener() {
             @Override
-            public void onClick(View view) {
+            public void onClick(View v) {
                 progressBar.setVisibility(View.VISIBLE);
-               String email;
-               String password;
-               //read email and password:
+                String email;
+                String password;
+                //read email and password:
                 email = String.valueOf(editTextEmail.getText());
                 password = String.valueOf(editTextPassword.getText());
 
                 if (TextUtils.isEmpty(email)){
-                    Toast.makeText(SignUp.this, "Please enter email", Toast.LENGTH_SHORT).show();
+                    Toast.makeText(getActivity(), "Please enter email", Toast.LENGTH_SHORT).show();
+                    progressBar.setVisibility(View.GONE);
                     return;
                 }
                 if (TextUtils.isEmpty(password)){
-                    Toast.makeText(SignUp.this, "Please enter password", Toast.LENGTH_SHORT).show();
+                    Toast.makeText(getActivity(), "Please enter password", Toast.LENGTH_SHORT).show();
+                    progressBar.setVisibility(View.GONE);
+                    return;
+                }
+                if (password.length() < 6){
+                    Toast.makeText(getActivity(), "Password should be at least 6 characters", Toast.LENGTH_SHORT).show();
+                    progressBar.setVisibility(View.GONE);
                     return;
                 }
+
                 mAuth.createUserWithEmailAndPassword(email, password)
                         .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                             @Override
                             public void onComplete(@NonNull Task<AuthResult> task) {
                                 progressBar.setVisibility(View.GONE);
                                 if (task.isSuccessful()) {
-                                    Toast.makeText(SignUp.this, "Account Created.",
+                                    Toast.makeText(getActivity(), "Account Created.",
                                             Toast.LENGTH_SHORT).show();
+                                    Intent intent = new Intent(getActivity(), Login.class);
+                                    startActivity(intent);
+                                    getActivity().finish();
                                 } else {
                                     // If sign in fails, display a message to the user.
-                                    Toast.makeText(SignUp.this, "Authentication failed.",
+                                    Toast.makeText(getActivity(), "Authentication failed.",
                                             Toast.LENGTH_SHORT).show();
                                 }
                             }
Index: app/src/main/res/layout/fragment_welcome.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/res/layout/fragment_welcome.xml b/app/src/main/res/layout/fragment_welcome.xml
--- a/app/src/main/res/layout/fragment_welcome.xml	(revision 4f3b9dcc605b445b297203dda541b5e2d925738d)
+++ b/app/src/main/res/layout/fragment_welcome.xml	(date 1690485122850)
@@ -7,10 +7,11 @@
     tools:context=".WelcomeScreen">
 
     <TextView
+        android:text="I am a..."
         android:id="@+id/textView"
-        android:layout_width="wrap_content"
+        android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        android:text="I am a..."
+        android:gravity="center"
         android:textSize="24sp"
         app:layout_constraintBottom_toTopOf="@+id/button_shopper"
         app:layout_constraintEnd_toEndOf="parent"
Index: app/src/main/java/com/example/shopeaze/Login.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/com/example/shopeaze/Login.java b/app/src/main/java/com/example/shopeaze/Login.java
--- a/app/src/main/java/com/example/shopeaze/Login.java	(revision 4f3b9dcc605b445b297203dda541b5e2d925738d)
+++ b/app/src/main/java/com/example/shopeaze/Login.java	(date 1690677547695)
@@ -1,13 +1,15 @@
 package com.example.shopeaze;
 
 import androidx.annotation.NonNull;
-import androidx.appcompat.app.AppCompatActivity;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
 
 import android.content.Intent;
 import android.os.Bundle;
 import android.text.TextUtils;
-import android.util.Log;
+import android.view.LayoutInflater;
 import android.view.View;
+import android.view.ViewGroup;
 import android.widget.Button;
 import android.widget.ProgressBar;
 import android.widget.TextView;
@@ -20,7 +22,7 @@
 import com.google.firebase.auth.FirebaseAuth;
 import com.google.firebase.auth.FirebaseUser;
 
-public class Login extends AppCompatActivity {
+public class Login extends Fragment {
 
     TextInputEditText editTextEmail, editTextPassword;
     Button buttonLogin;
@@ -31,37 +33,43 @@
     @Override
     public void onStart() {
         super.onStart();
-        // Check if user is alr signed in (non-null)
+        mAuth = FirebaseAuth.getInstance();
+        // Check if user is already signed in (non-null)
         FirebaseUser currentUser = mAuth.getCurrentUser();
         if(currentUser != null){
-            Intent intent = new Intent(getApplicationContext(), MainActivity.class);
+            Intent intent = new Intent(getActivity(), MainActivity.class);
             startActivity(intent);
-            finish();
+            getActivity().finish();
         }
     }
+
+    @Nullable
     @Override
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_login);
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.activity_login, container, false);
+    }
 
-        mAuth = FirebaseAuth.getInstance();
-        editTextEmail = findViewById(R.id.email);
-        editTextPassword = findViewById(R.id.password);
-        buttonLogin = findViewById(R.id.btn_login);
-        progressBar = findViewById(R.id.progressBar);
-        textView = findViewById(R.id.signUpNow);
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+
+        editTextEmail = view.findViewById(R.id.email);
+        editTextPassword = view.findViewById(R.id.password);
+        buttonLogin = view.findViewById(R.id.btn_login);
+        progressBar = view.findViewById(R.id.progressBar);
+        textView = view.findViewById(R.id.signUpNow);
         textView.setOnClickListener(new View.OnClickListener(){
             @Override
-            public void onClick(View view){
-                Intent intent = new Intent(getApplicationContext(), SignUp.class);
+            public void onClick(View v){
+                Intent intent = new Intent(getActivity(), SignUp.class);
                 startActivity(intent);
-                finish();
+                getActivity().finish();
             }
         });
 
         buttonLogin.setOnClickListener(new View.OnClickListener() {
             @Override
-            public void onClick(View view) {
+            public void onClick(View v) {
                 progressBar.setVisibility(View.VISIBLE);
                 String email;
                 String password;
@@ -70,11 +78,13 @@
                 password = String.valueOf(editTextPassword.getText());
 
                 if (TextUtils.isEmpty(email)){
-                    Toast.makeText(Login.this, "Please enter email", Toast.LENGTH_SHORT).show();
+                    Toast.makeText(getActivity(), "Please enter email", Toast.LENGTH_SHORT).show();
+                    progressBar.setVisibility(View.GONE);
                     return;
                 }
                 if (TextUtils.isEmpty(password)){
-                    Toast.makeText(Login.this, "Please enter password", Toast.LENGTH_SHORT).show();
+                    Toast.makeText(getActivity(), "Please enter password", Toast.LENGTH_SHORT).show();
+                    progressBar.setVisibility(View.GONE);
                     return;
                 }
 
@@ -84,21 +94,21 @@
                             public void onComplete(@NonNull Task<AuthResult> task) {
                                 progressBar.setVisibility(View.GONE);
                                 if (task.isSuccessful()) {
-                                    Toast.makeText(getApplicationContext(), "Login Successful", Toast.LENGTH_SHORT).show();
-                                    Intent intent = new Intent(getApplicationContext(), MainActivity.class);
+                                    Toast.makeText(getActivity(), "Login Successful", Toast.LENGTH_SHORT).show();
+                                    Intent intent = new Intent(getActivity(), MainActivity.class);
                                     startActivity(intent);
-                                    finish();
+                                    getActivity().finish();
 
                                 } else {
-                                    Toast.makeText(Login.this, "Authentication failed.",
+                                    //display message if email already exists (didnt do that yet):
+
+                                    //or, other issue:
+                                    Toast.makeText(getActivity(), "Authentication failed.",
                                             Toast.LENGTH_SHORT).show();
                                 }
                             }
                         });
-
             }
         });
-
-
     }
 }
\ No newline at end of file
